src/Raven.Server/Documents/DatabasesLandlord.cs

```diff
@@ -1056,11 +1056,11 @@ public bool UnloadDirectly(StringSegment databaseName, IdleDatabaseActivity idle
 
                 // DateTime should be only null in tests
                 if (idleDatabaseActivity is { DateTime: not null })
-                    _wakeupTimers.TryAdd(databaseName.Value, new Timer(
+                    _wakeupTimers.GetOrAdd(databaseName.Value, _ => new Timer(
```

Comment:
@garayx this should be AddOrUpdate or GetOrAdd? Should we remove timer here like in 'RescheduleNextIdleDatabaseActivity' when 'idleDatabaseActivity' is null?

Reply:
we already removed the timer few line above (in `ShouldContinueDispose()` on line 1028), so there should be no timer, so I would keep `TryAdd()` or use `AddOrUpdate()` to be on the safe side

we always remove the timer when `ShouldContinueDispose()` calling method few lines above, regardless of idleDatabaseActivity value.

Reply:
switched to AddOrUpdate